jubilant/_juju.py

```diff
@@ -205,6 +205,151 @@ def add_unit(
+        metadata_source: str | pathlib.Path | None = None,
+        model_default: Mapping[str, ConfigValue] | None = None,
+        no_switch: bool = False,
+        regions: str | None = None,
+        storage_pool: Mapping[str, str] | None = None,
+        to: str | None = None,
+    ) -> str:
+        """Initializes a cloud environment.
+
+        Args:
+            cloud: Name of cloud to bootstrap on. Used without arguments, bootstrap will work
+                in interactive mode and step you through the process of initializing a Juju
+                cloud environment. Initialization consists of creating a "controller" model
+                and provisioning a machine to act as controller.
+            controller: Name for the controller. If not provided, Juju will generate one.
+                Controller names may only contain lowercase letters, digits and hyphens, and
+                may not start with a hyphen. We recommend you call your controller
+                'username-region' e.g. 'fred-us-east-1'. See ``clouds`` for a list of clouds and
+                credentials. See ``regions`` for a list of available regions for a given cloud.
+            add_model: Name of an initial model to create on the new controller.
+            agent_version: Version of agent binaries to use for Juju agents.
+            auto_upgrade: If true, after bootstrap, upgrade to the latest patch release.
+            bootstrap_base: Specify the base of the bootstrap machine.
+            bootstrap_constraints: Specify bootstrap machine constraints, for example,
+                ``{'mem': '8G'}``. If used, its values will also apply to any future
+                controllers provisioned for high availability (HA).
+            bootstrap_image: Specify the image of the bootstrap machine (requires
+                bootstrap_constraints specifying architecture).
+            bootstrap_series: Specify the series of the bootstrap machine (deprecated, use
+                bootstrap_base).
+            build_agent: If true, build local version of agent binary before bootstrapping.
+            clouds: If true, print the available clouds which can be used to bootstrap a
+                Juju environment.
+            config: Controller configuration options. Model config keys only affect the
+                controller model.
+            constraints: Set model constraints, for example, ``{'mem': '8G', 'cores': '4'}``.
+                If used, its values will be set as the default constraints for all future
+                workload machines in the model, exactly as if the constraints were set with
+                ``juju set-model-constraints``.
+            controller_charm_channel: The Charmhub channel to download the controller charm
+                from (if not using a local charm).
+            controller_charm_path: Path to a locally built controller charm.
+            credential: Credentials to use when bootstrapping.
```

Comment:
I believe this is the name of a previously added credential. So we should probably clarify, like we do in the `add_model` docstring:

```suggestion
            credential: Name of cloud credential to use when bootstrapping.
```